beamer/tests/agent/test_request.py

```diff
@@ -377,3 +377,39 @@ def test_event_fetcher_progress_without_events(agent, direction):
                 != ape.chain.blocks[-1].number
             ):
                 sleeper.sleep(0.1)
+
+
+def test_handling_reorgs(config, direction, token, request_manager):
+    requester, target = alloc_accounts(2)
+    confirmation_blocks = 2
+    config.confirmation_blocks["l2a"] = confirmation_blocks
+    agent = Agent(config)
+    agent.start()
+
+    snapshot = ape.chain.snapshot()
+    ape.chain.mine(confirmation_blocks + 1)
+    request_id_1 = make_request(request_manager, token, requester, target, 1)
+
+    assert ape.chain.blocks[-1].number
```

Comment:
We're checking whether the block number is zero? Is that intentional?

Reply:
it is mypy again and the reason is actually this expression return type it is int | None so i am making sure that this is not None

Reply:
> it is mypy again and the reason is actually this expression return type it is int | None so i am making sure that this is not None

Ah, mypy again. :woozy_face: 
But then, please make it explicit that you are testing against `None`:

```python
    assert ape.chain.blocks[-1].number is not None
```